airbyte-integrations/connectors/source-pipedrive/source_pipedrive/streams.py

```diff
@@ -0,0 +1,192 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from abc import ABC
+from typing import Any, Iterable, Mapping, MutableMapping, Optional
+
+import pendulum
+import requests
+from airbyte_cdk.sources.streams.http import HttpStream
+
+PIPEDRIVE_URL_BASE = "https://api.pipedrive.com/v1/"
+
+
+class PipedriveStream(HttpStream, ABC):
+    url_base = PIPEDRIVE_URL_BASE
+    primary_key = "id"
+    data_field = "data"
+
+    def __init__(self, api_token: str, **kwargs):
+        super().__init__(**kwargs)
+        self._api_token = api_token
+
+    def path(self, **kwargs) -> str:
+        class_name = self.__class__.__name__
+        return f"{class_name[0].lower()}{class_name[1:]}"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        return {"api_token": self._api_token}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        records = response.json().get(self.data_field) or []
+        yield from records
+
+
+class PaginatedPipedriveStream(PipedriveStream):
+    limit = 50
```

Comment:
```suggestion
    page_size = 50
```

I find this name is unambiguous, WDYT?